django/utils/html.py

```diff
@@ -268,24 +268,40 @@ def unescape(text, trail):
             trail = ''
         return text, unescaped, trail
 
+    def trim_trailing_punctuation(lead, middle, trail):
```

Comment:
Some docstrings for these functions would help. The variables aren't named such that their purpose is obvious to me.